b_cfn_sagemaker_endpoint/refresh/source/index.py

```diff
@@ -0,0 +1,68 @@
+import json
+import os
+import time
+from typing import Any, Dict
+import boto3
+
+
+class Counter:
+    def __init__(self, start: int = 0, step: int = 1):
+        self.value = start
+        self.__step = step
+
+    def next(self) -> int:
+        self.value += self.__step
+        return self.value
+
+
+invocation_counter = Counter()
+
+
+def handler(event: Dict[str, Any], context: Any) -> None:
+    invocation_number = invocation_counter.next()
+    print('Current invocation number:', invocation_number)
+    print(f'S3 event received: {json.dumps(event)}')
+
+    wait_time = float(os.environ['WAIT_TIME'])
+    endpoint_name = os.environ['SAGEMAKER_ENDPOINT_NAME']
+    endpoint_config_a_name = os.environ['SAGEMAKER_ENDPOINT_CONFIG_A_NAME']
+    endpoint_config_b_name = os.environ['SAGEMAKER_ENDPOINT_CONFIG_B_NAME']
+    print(
+        'Using the following environment variables: '
+        f'{wait_time=} '
+        f'{endpoint_name=} '
+        f'{endpoint_config_a_name=} '
+        f'{endpoint_config_b_name=} '
+    )
+
+    # Wait for any other bucket objects to be uploaded.
+    print('Waiting on standby...')
```

Comment:
Why are you waiting for some random number? You can ensure that only a single instance can be run at a time, all others would get throttled.

Reply:
The duration for which lambda will wait will be passed by the user. 

As for the need to wait for a given time, say 60s:  I must wait before calling `update_endpoint` function until all files are uploaded to s3 bucket.  Since multiple files will not be uploaded to the bucket at the same time, premature `update_endpoint` call might fail to pull all the required files from s3.